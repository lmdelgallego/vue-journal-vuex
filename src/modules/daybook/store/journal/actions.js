/*
export const myAction = async ({commit}) => {

}
*/

import journalApi from '@/api/journalApi';

export const loadEntries = async ({commit}) => {
  const {data} = await journalApi.get('/entries.json');
  const entries = [];
  for (let id of Object.keys(data))  {
      entries.push({id, ...data[id]})
  }
  commit('setEntries', entries);
}

export const createEntry = async ({commit},entry) => {
  console.log(entry);
  console.log(commit)
  // await journalApi.put(`/${entry.id}.json`, entry );

}

export const updateEntry = async ({commit}, entry) => {
  const {date, picture, text} = entry;
  const dataToSave = {date,picture,text};
  await journalApi.put(`/entries/${entry.id}.json`, dataToSave );
  commit('updateEntries', {...entry});
}
