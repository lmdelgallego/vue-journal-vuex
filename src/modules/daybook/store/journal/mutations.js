/*
export const myMutation = (state) => {

}
* */

export const setEntries = (state, entries) => {
  state.entries = [...state.entries, ...entries];
  state.isLoading = false;
}

export const updateEntries = (/*state*/) => {

}
export const addEntries = (/*state*/) => {

}
