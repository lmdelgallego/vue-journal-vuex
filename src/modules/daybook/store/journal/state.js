export default () => ({
  isLoading: true,
  entries: [
    {
      id: '1',
      date: new Date().toDateString(),
      text:
        'Anim anim consequat ea sit excepteur consectetur minim Lorem amet nostrud. Id ea ut occaecat officia. Velit non anim tempor quis nulla id ea. Exercitation minim esse eu enim ut nisi incididunt Lorem. Minim in duis commodo consectetur sint anim ut consequat. Duis laborum consectetur mollit dolor culpa minim.',
      picture: null,
    },
    {
      id: '2',
      date: new Date().toDateString(),
      text:
        'Magna in enim cupidatat et tempor. Adipisicing duis culpa est mollit veniam id fugiat Lorem aliquip. Laboris deserunt amet occaecat reprehenderit.',
      picture: null,
    },
    {
      id: '3',
      date: new Date().toDateString(),
      text:
        'Esse duis aliqua nisi aliqua do mollit tempor adipisicing occaecat do laboris ad. Nostrud et deserunt consectetur commodo Lorem deserunt ipsum. Cillum magna adipisicing reprehenderit quis adipisicing ad tempor est aliquip quis veniam sint magna ullamco. Duis non elit cupidatat eiusmod magna adipisicing minim consectetur exercitation dolor culpa occaecat esse nisi. Velit Lorem excepteur esse anim nulla incididunt. Qui tempor voluptate esse cillum elit consectetur nisi eu eu dolore.',
      picture: null,
    },
    {
      id: '4',
      date: new Date().toDateString(),
      text:
        'EVeniam dolor enim amet cupidatat cillum officia reprehenderit pariatur. Proident adipisicing mollit culpa eu. Velit tempor enim in nostrud commodo esse eu proident dolor proident qui occaecat officia. Do in ex tempor nulla ex in. Minim Lorem ad commodo elit ea id voluptate ut ut.',
      picture: null,
    },
    {
      id: '5',
      date: new Date().toDateString(),
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicin',
      picture: null,
    },
  ],
});
