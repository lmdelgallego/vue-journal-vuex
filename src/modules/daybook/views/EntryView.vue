<template>
  <div class="entry-title d-flex justify-content-between p-2">
    <div>
      <span class="text-success fs-3 fw-bold">15</span>
      <span class="mx-1 fs-3">Julio</span>
      <span class="mx-2 fs-4 fw-light">2021, jueves</span>
    </div>

    <div>
      <button class="btn btn-danger mx-2">
        Borrar <i class="fa fa-trash-alt"></i>
      </button>

      <button class="btn btn-primary">
        Subir foto <i class="fa fa-upload"></i>
      </button>
    </div>
  </div>
  <hr />
  <div class="d-flex flex-column px-3 h-75">
    <textarea placeholder="Que sucedio hoy"></textarea>
  </div>

  <img
    src="https://i.blogs.es/94ad8d/inside-plane-hanger-looking-down/1366_2000.jpg"
    alt="entry-picture"
    class="img-thumbnail"
  />

  <Fab icon="fa-save" />
</template>

<script>
import { defineAsyncComponent } from "@vue/runtime-core";
import {mapGetters} from 'vuex';

export default {
  props:{
    id: {
      type: String,
      required: true
    }
  },
  components: {
    Fab: defineAsyncComponent(() => import("../components/Fab.vue")),
  },
  computed: {
    ...mapGetters('journal', ['getEntryById'])
  },
  methods:{
    loadEntry(){
      const entry = this.getEntryById(this.id);
      console.log(entry)
    }
  },
  created() {
    this.loadEntry();
  }
};
</script>

<style lang="scss" scoped>
textarea {
  font-size: 1.5rem;
  border: none;
  height: 100%;
  &:focus {
    outline: none;
  }
}
img {
  width: 200px;
  position: fixed;
  bottom: 150px;
  right: 20px;
  box-shadow: 0 5px 10px rgba($color: #000000, $alpha: 0.2);
}
</style>
