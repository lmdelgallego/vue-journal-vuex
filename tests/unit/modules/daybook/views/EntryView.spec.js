import { createStore } from "vuex";
import { shallowMount } from "@vue/test-utils";
import journal from '@/modules/daybook/store/journal';
import EntryView from '@/modules/daybook/views/EntryView.vue';
import { testJournalEntry, testJournalState } from "../../../mocks/test-journal-state";


const createVuexStore = (initialState) => createStore({
  modules: {
    journal: {
      ...journal,
      state: {...initialState}
    },
  }
});

const mockRoute = {
  push: jest.fn(),
};


describe('EntryView', () => {

  const store = createVuexStore(testJournalState);
  let wrapper ;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryView, {
      props: {
        id: testJournalEntry.id,
      },
      global: {
        mocks: {
          $router: mockRoute
        },
        plugins: [store],
      }
    });
  });
  test("should be out user if id don't exist", () => {
    const wrapper = shallowMount(EntryView, {
      props: {
        id: 'does not exist'
      },
      global: {
        mocks: {
          $router: mockRoute
        },
        plugins: [store],
      }
    });

    expect(mockRoute.push).toBeCalled();
    expect(mockRoute.push).toBeCalledWith({name: 'daybook-no-entry'});

  });

  test("should be show entry correctly", ()=> {
    expect(wrapper.html()).toMatchSnapshot();
    expect(mockRoute.push).not.toBeCalled();
  })
});