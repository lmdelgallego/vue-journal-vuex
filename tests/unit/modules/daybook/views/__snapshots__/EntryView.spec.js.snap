// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryView should be show entry correctly 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">29</span><span class="mx-1 fs-3">Octubre</span><span class="mx-2 fs-4 fw-light">2021, Viernes</span></div>
  <div><input type="file" accept="image/png, image/jpeg, image/gif" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="Que sucedio hoy"></textarea></div>
<div class="entry-picture"><img src="https://res.cloudinary.com/alucardluis/image/upload/v1635539273/t0hruhdohpmdmmcz29nx.png" alt="entry-picture" class="img-thumbnail"></div>
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
